src/annotator/plugin/test/bucket-bar-test.js

```diff
@@ -0,0 +1,227 @@
+import $ from 'jquery';
+import BucketBar from '../bucket-bar';
+import { $imports } from '../bucket-bar';
+
+// Return DOM elements for non-empty bucket indicators in a `BucketBar`.
+const nonEmptyBuckets = function (bucketBar) {
+  const buckets = bucketBar.element[0].querySelectorAll(
+    '.annotator-bucket-indicator'
+  );
+  return Array.from(buckets).filter(bucket => {
+    const label = bucket.querySelector('.label');
+    return parseInt(label.textContent) > 0;
+  });
+};
+
+const createMouseEvent = function (type, { ctrlKey, metaKey } = {}) {
+  // In a modern browser we could use `new MouseEvent` constructor and pass
+  // `ctrlKey` and `metaKey` via the init object.
+  const event = new Event(type);
+  event.ctrlKey = Boolean(ctrlKey);
+  event.metaKey = Boolean(metaKey);
+  return event;
+};
+
+describe('BucketBar', function () {
+  let fakeAnnotator;
+  let fakeBucketUtil;
+
+  beforeEach(() => {
+    fakeAnnotator = {
+      anchors: [],
+      selectAnnotations: sinon.stub(),
+    };
+
+    fakeBucketUtil = {
+      findClosestOffscreenAnchor: sinon.stub(),
+      constructPositionPoints: sinon
+        .stub()
+        .returns({ above: [], below: [], points: [] }),
+      buildBuckets: sinon.stub().returns([]),
+    };
+
+    $imports.$mock({
+      '../util/buckets': fakeBucketUtil,
+    });
+  });
+
+  afterEach(() => {
+    $imports.$restore();
+  });
+
+  const createBucketBar = function (options) {
+    const element = document.createElement('div');
+    return new BucketBar(element, options || {}, fakeAnnotator);
+  };
+
+  // Create a fake anchor, which is a combination of annotation object and
+  // associated highlight elements.
+  const createAnchor = () => {
+    return {
+      annotation: { $tag: 'ann1' },
+      highlights: [document.createElement('span')],
+    };
+  };
+
+  context('when a bucket is clicked', () => {
+    let bucketBar;
+
+    beforeEach(() => {
+      bucketBar = createBucketBar();
+      // Create fake anchors and render buckets.
+      const anchors = [createAnchor()];
+
+      fakeBucketUtil.buildBuckets.returns({
+        index: [250],
+        buckets: [[anchors[0]]],
+      });
+
+      bucketBar.annotator.anchors = anchors;
+      bucketBar._update();
+    });
+
+    it('selects the annotations', () => {
+      // Click on the indicator for the non-empty bucket.
+      const bucketEls = nonEmptyBuckets(bucketBar);
+      assert.equal(bucketEls.length, 1);
+      bucketEls[0].dispatchEvent(createMouseEvent('click'));
+
+      const anns = bucketBar.annotator.anchors.map(anchor => anchor.annotation);
+      return assert.calledWith(
```

Comment:
```suggestion
      assert.calledWith(
```

This is a common issue when decaffeinating tests. The same applies for other occurrences of `return assert...` in this file.

Reply:
This has been done.